### PR TITLE
Typings

### PR DESCRIPTION
**Pull request by nathanmarks**
Adds typings.

For some reason, I had linting failing on a few specs because they weren't named `.spec.js` like the other tests... I renamed them `.spec.js` so had to rewrite the snapshots for them as well.

### CODE REVIEW

File: test/typescript/FooRefTest.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable */
```
Comment (mmarcuccio) [inline test/typescript/FooRefTest.tsx]: Do we still need these files in this test directory?

Comment (nathanmarks) [reply test/typescript/FooRefTest.tsx]: I was using these to test some typings -- i think it's useful to have them. But I can kill them if you want.

File: src/components/Forms/Form.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+type FormComponent = React.Component<FormComponentInjectedProps>
+
+export interface FormProps {
+  /** Form html chilren */
+  children?: React.ReactNode
+
+  /** HTML form attributes */
+  formProps?: React.ComponentPropsWithoutRef<'form'>
```
[the remark this answers is not in the document]
Comment (nathanmarks) [reply src/components/Forms/Form.d.ts]: Good catch!!!!

File: test/typescript/FooRefTest.tsx (hunk 1 above)
Comment (mmarcuccio) [reply test/typescript/FooRefTest.tsx]: I'm fine either way.

File: .eslintrc.js [hunk 3]
```diff
@@ -27,6 +27,11 @@ module.exports = {
     'react/forbid-prop-types': [WARN, { forbid: 'any' }],
     'react/jsx-filename-extension': OFF,
     'react/jsx-no-literals': OFF,
+
+    'import/no-extraneous-dependencies': [
+      'error',
+      { devDependencies: ['scripts/**/*.js', '**/*.spec.js'] },
```
Comment (mmarcuccio) [inline .eslintrc.js]: We ended up doing this in customers too.

**Pull request merged**